### PR TITLE
Add log for random exception in Linux GPU Test Stage. 

### PR DESCRIPTION
**Pull request by mszhanyi**
### Description
1. check GPU status in docker
2. use stages to make test stage can leverage existing building artifacts

### Motivation and Context
To investigate the root cause of the random exception
`CUDA failure 100: no CUDA-capable device is detected`

### CODE REVIEW

**Review by snnn — APPROVED** [review]

**Comment by tianleiwu** [comment]
Tests are skipped (only 10 seconds) ? 
image

https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1299002&view=logs&j=95838853-166c-5f4e-049e-ad4aefc5dff3&t=05c1745b-30a3-5ee0-30ea-d4196d3209bd&l=26
image
The machine installed CUDA 12, does that impact the testing in docker (since the build is for CUDA 11.8), and memory only 4GB?

Maybe add a line to run nvidia-smi inside the docker?

**Comment by mszhanyi** [comment]
> Tests are skipped (only run 25 seconds) ? image
> 
> https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1299002&view=logs&j=95838853-166c-5f4e-049e-ad4aefc5dff3&t=05c1745b-30a3-5ee0-30ea-d4196d3209bd&l=26 image The machine installed CUDA 12, does that impact the testing in docker (since the build is for CUDA 11.8)? GPU memory only 4GB?



> Tests are skipped (only run 25 seconds) ? image
> 
> https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1299002&view=logs&j=95838853-166c-5f4e-049e-ad4aefc5dff3&t=05c1745b-30a3-5ee0-30ea-d4196d3209bd&l=26 image The machine installed CUDA 12, does that impact the testing in docker (since the build is for CUDA 11.8)? GPU memory only 4GB?

nvidia-smi shows the CUDA driver version. As long as the CUDA runtime version in docker is 11.8, it doesn't impact the testing.
Since A series, nvidia GPU can be partitioned. 
4GB is enough for all unit tests and model tests in model zoo. It could improve the utilization of limited GPU resources.
https://learn.microsoft.com/en-us/azure/virtual-machines/nva10v5-series
We have set up some pools for models need big GPU memory.

File: tools/ci_build/github/azure-pipelines/linux-gpu-ci-pipeline.yml [hunk 1]
```diff
@@ -61,163 +72,201 @@ variables:
...
+            $(Repository) \
+            /bin/bash -c '
+              nvidia-smi; \
+              /sbin/ldconfig -N -v $(sed "s/:/ /" <<< $LD_LIBRARY_PATH) 2>/dev/null | grep libcudart; \
+              /sbin/ldconfig -N -v $(sed "s/:/ /" <<< $LD_LIBRARY_PATH) 2>/dev/null | grep libcudnn; \
+              /sbin/ldconfig -N -v $(sed "s/:/ /" <<< $LD_LIBRARY_PATH) 2>/dev/null | grep libnvinfer; \
+              whereis cudnn; \
+              whereis cuda; \
+              cat /usr/local/cuda/include/cudnn.h | grep CUDNN_MAJOR -A 2; \
+              cat /usr/include/x86_64-linux-gnu/cudnn_v*.h | grep CUDNN_MAJOR -A 2; \
```
Comment (tianleiwu) [inline tools/ci_build/github/azure-pipelines/linux-gpu-ci-pipeline.yml:233]:
```suggestion
              /sbin/ldconfig -N -v $(sed "s/:/ /" <<< $LD_LIBRARY_PATH) 2>/dev/null | grep libcudart.so; \
              /sbin/ldconfig -N -v $(sed "s/:/ /" <<< $LD_LIBRARY_PATH) 2>/dev/null | grep libcudnn.so; \
              /sbin/ldconfig -N -v $(sed "s/:/ /" <<< $LD_LIBRARY_PATH) 2>/dev/null | grep libnvinfer.so; \
              cat /usr/local/cuda/include/cuda.h | grep -m1 CUDA_VERSION; \
              cat /usr/include/cudnn_version.h | grep CUDNN_MAJOR -m1 -A 2; \
```

**Comment by tianleiwu** [comment]
> Since A series, nvidia GPU can be partitioned. 4GB is enough for all unit tests and model tests in model zoo. 

Not sure whether the issue is related to MIG.

**Review by tianleiwu — DISMISSED** [review]

**Review by snnn — DISMISSED** [review]

**Review by tianleiwu — APPROVED** [review]

**Review by snnn — APPROVED** [review]

**Pull request merged**